ui/job-view/headerbars/WatchedRepo.jsx

```diff
@@ -137,9 +137,15 @@ class WatchedRepo extends React.Component {
     return (
       <ButtonGroup>
         <Button
-          onClick={() => {
+          href={updateRepoParams(watchedRepo)}
+          onClick={(event) => {
             pushRoute({ search: updateRepoParams(watchedRepo) });
             this.updateTreeStatus(true);
+            event.preventDefault();
```

Comment:
I don't think this is the right approach. We need to very careful about using semantic elements to make this inclusive for people who use screen readers (please take a look at the [accessibility](https://treeherder.readthedocs.io/accessibility.html#accessibility) section in our docs).

Additionally, I don't think we want to change this behavior completely. Opening a tab in a new link might be a personal preference rather something the majority wants or expects :) 

Can you try to find a solution that utilizes event handlers (based on key press or mouse click, as Aryx mentioned middle mouse click in this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1674605))?

Reply:
utilizing event doesn't respect the user's expectation/configuration about how things work, and I think we shouldn't go that way.

then, in term of semantics, "watched repos" are links for me, as clicking it changes the URL at the same time as changing the content, and also reloading the URL shows the same content.
and actually, they were links until at least 2 years ago (I don't remember when it was changed to buttons tho), that's the reason why middle click opened a new tab before, as mentioned in the bug.

then, if they're changed to links, they work as the following:
  * if the link is primary-clicked, what it does is basically "load the repository's jobs page", that's how usual link works. But instead of performing browser's navigation, it modify the page content to achieve the same result without navigation, that's what the onclick handler does.  so, even without onclick handler, the result is same after the navigation
  * if the link is middle-clicked (with the configuration that middle-click opens the link in a new tab), the repository's page is opened in a new tab
  * if the link is secondary-clicked, context menu for the link is shown, that provides users more control about what to do for the link

in my opinion that doesn't break accessibility requirement.
(maybe some attributes need to be updated tho, haven't checked aria-* attributes)
https://treeherder.readthedocs.io/accessibility.html#interactive-elements
> Interactive elements
> You can also choose a HTML <a> element.

Reply:
> utilizing event doesn't respect the user's expectation/configuration about how things work, and I think we shouldn't go that way.

sorry, my comment was unclear.

(1) we cannot assume certain user input is intended for opening a new tab

On some system, middle-click is assigned for paste (e.g. https://searchfox.org/mozilla-central/rev/21a9b72545da06681db97c4b3a2a6be761f4aae5/modules/libpref/init/all.js#3286-3287).
So, hard-coding "middle-click => open a new tab" doesn't follow the system's behavior.

(2) we cannot assume certain API call opens a new tab

if we implement the "open a new tab" as website's feature, instead of browser's behavior, it needs to be implemented with Web API calls,
but there's no API that guarantees "open a new tab" behavior.

So the result is user's environment dependent, and we cannot be sure what the middle-click handler does (we could say "in some/common environment, this opens a new tab" tho...).
also that's hard to be documented.

Reply:
A button that behaves like a tab and doesn't actually navigate to a new view - regardless of whether it changes a query string - that is debatable whether a button or link should be used. I don't agree with your opinion. However, I generally don't have a problem with changing these back to links as long they don't open a new tab by default and don't break accessibility or cause confusion for our screen reader users.

What I meant by using the correct semantic element, is that your code change is incorrect because you added an `href` to a button which is incorrect. If its to be a link then it needs an anchor tag with an href. Also, you have `updateRepoParams(watchedRepo)` twice, which is probably unnecessary. If we want to ensure, when navigating, that the newest entry is on the react router history stack it might be better to use the react router [link](https://v5.reactrouter.com/web/api/Link) element.

Additionally, once you make changes please confirm whether the onClick action is actually necessary and that you have verified that there aren't any undesireable side effects due to changing to a link (duplicate API calls or missing state updates) via manual testing.

Reply:
Also, to clear up some confusion when I made this comment "Opening a tab in a new link might be a personal preference rather something the majority wants or expects " I initially thought you wanted the link to open a new window or tab **by default** rather than having that option by right or middle click. I realize I had misread your bugzilla bug and that's not your intention.

Reply:
> What I meant by using the correct semantic element, is that your code change is incorrect because you added an `href` to a button which is incorrect. If its to be a link then it needs an anchor tag with an href.

In which layer is the "semantic element" about?
The reactstrap's element, or the resulting raw HTML element?

If if the "semantic element" is about reactstrap's element layer, and using `href` for `<Button>` is prohibited there, then I'll look for something else than `<ButtonGroup>`+`<Button>`.
but in my opinion this is not about accessibility, but just convention, given they're not exposed to DOM or accessibility API.
of course, if that is the convention, I'm fine following it.

If the "semantic element" is about the resulting raw HTML element (that's exposed to screen reader application), this change should be fine.
By adding `href` attribute to reactstrap's `<Button>` element, it uses `<a>` element for the resulting HTML element:
https://github.com/reactstrap/reactstrap/blob/37a63c673269bcd6eb3a7331583afa94113b73f6/src/Button.js#L74-L76
```
class Button extends React.Component {
...
  render() {
...
    if (attributes.href && Tag === 'button') {
      Tag = 'a';
    }
...
    return (
      <Tag
```

So at least passing `href` attribute there is not incorrect in term of API usage, and the resulting HTML element matches semantics.

> Also, you have `updateRepoParams(watchedRepo)` twice, which is probably unnecessary.

Do you mean `onClick` handler should get the query from `href` attribute?
I thought it's better directly calling `updateRepoParams(watchedRepo)` there instead of retrieving raw DOM attribute, given `updateRepoParams` has no side effect.

> Additionally, once you make changes please confirm whether the onClick action is actually necessary and that you have verified that there aren't any undesireable side effects due to changing to a link (duplicate API calls or missing state updates) via manual testing.

In term of the resulting page, `onClick` isn't necessary.
`onClick` is kept there to avoid browser's nagivation (loading the page from scratch), while achieving the same result.

Reply:
okay, I realized we can pass `tag={Link} ` to `Button` to keep the styling while using `Link` that doesn't require `onClick`.
Thank you for your suggestion.

Reply:
No, that isn't what I was suggesting :) I meant to try using the react router dom <Link> element for navigation instead of the Button to see if that achieves the goal. There is no need to keep the reactstrap button element - we are overriding the styling with the classNames we pass in (if you go to the reactstrap docs page and look at the default styling you will see what I mean). Please just remove use of the Button element entirely and use either an anchor tag or the react router Link. 


> In term of the resulting page, onClick isn't necessary.
onClick is kept there to avoid browser's nagivation (loading the page from scratch), while achieving the same result.

I don't follow. At what point does the onclick occur - before or after we are navigated to the same page with only one query parameter changed?

Reply:
can you tell me the basic rule about reactstrap usage in the code?
I was thinking it's for styling, and they're apart from semantics, and I thought it's better keeping the styling structure.
also, what's the problem with `<Button tag={Link}>`?

> if you go to the reactstrap docs page and look at the default styling you will see what I mean

which page is it?

> Please just remove use of the Button element entirely and use either an anchor tag or the react router Link.

do we still keep using `<ButtonGroup>` ?

> > In term of the resulting page, onClick isn't necessary.
> > onClick is kept there to avoid browser's nagivation (loading the page from scratch), while achieving the same result.
> 
> I don't follow. At what point does the onclick occur - before or after we are navigated to the same page with only one query parameter changed?

with raw `<a>`, without `onClick`, clicking it causes nagivation.
`onClick` there was to suppress the nagivation, so it happens before the navigation, and instead of navigation, modifies the page content and also the query parameter.

anyway, if we use `Link`, this isn't necessary at all.

Reply:
maybe I'm still misunderstanding.
isn't `<Link>` what you mean by "react router dom element" ?

Reply:
Yes :) And it sounds like that might be the best option (just using it alone, not passed to the Button element) since it avoids the need to use onClick (probably because its just adding to the history props).

Reply:
Also, please don't squash your commits when you make changes. It makes it easier for me to see whats changed. I'll squash all commits when this pr gets merged.

Reply:
Can you please answer all my questions above?

Reply:
> can you tell me the basic rule about reactstrap usage in the code? I was thinking it's for styling, and they're apart from semantics, and I thought it's better keeping the styling structure. also, what's the problem with `<Button tag={Link}>`?
> 

I'd rather you use the react-router-dom Link element because its less confusing if someone in the future looks at that code, rather than seeing a button with a link tag passed to it. And as I said, I don't think we're using reacstrap styling. If you go to https://reactstrap.github.io/?path=/docs/components-button--button, you'll see what a default reactstrap button looks like. 

> do we still keep using `<ButtonGroup>` ?
If it's no longer a button, let's remove that

Reply:
okay, that makes sense, and it sounds reasonable to drop reactstrap there.
thank you for answering :)
I will update the patch shortly.